pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -67,6 +67,35 @@ var currentObjects = []client.Object{
 	&admissionregistrationv1beta1.MutatingWebhookConfiguration{TypeMeta: typeMeta(admissionregistrationv1beta1.SchemeGroupVersion, "MutatingWebhookConfiguration")},
 }
 
+func cloneObject(from client.Object) (client.Object, error) {
+	switch t := from.(type) {
+	case *rbacv1.ClusterRole:
+		return t.DeepCopyObject().(*rbacv1.ClusterRole), nil
+	case *rbacv1.ClusterRoleBinding:
+		return t.DeepCopyObject().(*rbacv1.ClusterRoleBinding), nil
+	case *storagev1beta1.CSIDriver:
+		return t.DeepCopyObject().(*storagev1beta1.CSIDriver), nil
+	case *appsv1.DaemonSet:
+		return t.DeepCopyObject().(*appsv1.DaemonSet), nil
+	case *rbacv1.Role:
+		return t.DeepCopyObject().(*rbacv1.Role), nil
+	case *rbacv1.RoleBinding:
+		return t.DeepCopyObject().(*rbacv1.RoleBinding), nil
+	case *corev1.Secret:
+		return t.DeepCopyObject().(*corev1.Secret), nil
+	case *corev1.Service:
+		return t.DeepCopyObject().(*corev1.Service), nil
+	case *corev1.ServiceAccount:
+		return t.DeepCopyObject().(*corev1.ServiceAccount), nil
+	case *appsv1.StatefulSet:
+		return t.DeepCopyObject().(*appsv1.StatefulSet), nil
+	case *admissionregistrationv1beta1.MutatingWebhookConfiguration:
+		return t.DeepCopyObject().(*admissionregistrationv1beta1.MutatingWebhookConfiguration), nil
+	default:
+		return nil, fmt.Errorf("cannot clone client.Object of type %T", from)
+	}
+}
```

Comment:
It would be nicer to avoid this. But `client.Object.DeepCopyObject` returns a `runtime.Object` which cannot be used for `client.Client.Patch` because it is not a `client.Object`.